e2e/helm_charts_test.go

```diff
@@ -0,0 +1,166 @@
+package e2e
+
+import (
+	"errors"
+	"strings"
+
+	shell "github.com/elastic/metricbeat-tests-poc/cli/shell"
+
+	"github.com/cucumber/godog"
+	log "github.com/sirupsen/logrus"
+)
+
+func init() {
+	binaries := []string{
+		"kubectl",
+		"helm",
+	}
+
+	shell.CheckInstalledSoftware(binaries)
+}
+
+// HelmChartTestSuite represents a test suite for a helm chart
+//nolint:unused
+type HelmChartTestSuite struct {
+	Name    string // the name of the chart
+	Version string // the helm chart version for the test
+}
+
+// getFullName returns the name plus version, in lowercase, enclosed in quotes
+func (ts *HelmChartTestSuite) getFullName() string {
+	return strings.ToLower("'" + ts.Name + "-" + ts.Version + "'")
+}
+
+func (ts *HelmChartTestSuite) delete() error {
+	args := []string{
+		"delete", ts.Name,
+	}
+
+	output, err := shell.Execute(".", "helm", args...)
+	if err != nil {
+		return err
+	}
+	log.WithFields(log.Fields{
+		"output": output,
+		"name":   ts.Name,
+	}).Debug("Chart deleted")
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) addElasticRepo() error {
+	elasticHelmChartsURL := "https://helm.elastic.co"
+
+	// use chart as application name
+	args := []string{
+		"repo", "add", "elastic", elasticHelmChartsURL,
+	}
+
+	output, err := shell.Execute(".", "helm", args...)
+	if err != nil {
+		return err
+	}
+	log.WithFields(log.Fields{
+		"output": output,
+		"name":   "elastic",
+		"url":    elasticHelmChartsURL,
+	}).Debug("Elastic Helm charts added")
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) install(chart string) error {
+	ts.Name = chart
+
+	elasticChart := "elastic/" + ts.Name
+
+	// use chart as application name
+	args := []string{
+		"install", ts.Name, elasticChart,
+	}
+
+	output, err := shell.Execute(".", "helm", args...)
+	if err != nil {
+		return err
+	}
+	log.WithFields(log.Fields{
+		"output": output,
+		"name":   ts.Name,
+		"chart":  elasticChart,
+	}).Debug("Chart installed")
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) aClusterIsRunning() error {
+	return nil
+}
+
+func (ts *HelmChartTestSuite) elasticsHelmChartIsInstalled(chart string) error {
+	ts.install(chart)
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) podsManagedByDaemonSet() error {
+	args := []string{
+		"get", "daemonsets", "--namespace=default", "-l", "app=" + ts.Name + "-" + ts.Name, "-o", "jsonpath='{.items[0].metadata.labels.chart}'",
```

Comment:
@kuisathaverat is this enough to check that a DaemonSet exists for the chart?

Reply:
This only ensures that the DaemonSet is deployed, but the DaemonSet pods can be in fail, so you have to check if the number of scheduled pods is the same as the pods ready(maybe is redundant) and the pods available.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#daemonset-v1-apps


```
kubectl get -n kube-system daemonset metricbeat -o jsonpath='{.status.currentNumberScheduled}'
kubectl get -n kube-system daemonset metricbeat -o jsonpath='{.status.numberAvailable}'
kubectl get -n kube-system daemonset metricbeat -o jsonpath='{.status.numberReady}'
```

Reply:
Wouldn't it mean we are checking that k8s is doing its job? 🤔

Reply:
no, you are checking the configuration settings of the pods are correct and can deploy